hotkeys.py

```diff
@@ -44,47 +44,89 @@ def load(self):
 		self.applyHotkeys()
 	#enddef
 
-	def register(self, seq, cb):
+	def _bind(self, seq, cb):
+		''' Binds a hotkey to a callback function.
+
+		seq can either be a QKeySequence or a string parseable by QKeySequence
+		eg "ctrl+shift+k" '''
 		try: seq = QKeySequence(seq)
 		except: pass
+
 		t = seqToTuple(seq)
-		log.debug('Registering hotkey {} => {}'.format(t, cb))
-		self.hk.register(t, callback=lambda e:cb())
+		log.debug('Binding hotkey {} => {}'.format(t, cb))
+		
+		try: self.hk.register(t, callback=lambda e:cb())
+		except SystemRegisterError: pass # log.exception('Failed to bind hotkey {}'.format(seq.toString()))
+		else: return True
+
+		return False
 	#enddef
 
-	def unregister(self, seq):
+	def _unbind(self, seq, quiet=False):
+		''' Removes a hotkey binding. If Quiet is true, will not warn if the binding doesn't exist '''
 		try: seq = QKeySequence(seq)
 		except: pass
+
 		t = seqToTuple(seq)
-		log.debug('Removing hotkey {}'.format(t))
-		self.hk.unregister(t)
+		log.debug('Unbinding hotkey {}'.format(t))
+
+		try: self.hk.unregister(t)
+		except InvalidKeyError: 
+			if not quiet: log.warn('Tried to unbind nonexistent hotkey {}'.format(seq.toString()))
+		except UnregisterError as e: log.exception('Failed to unbind hotkey {}'.format(seq.toString()))
+		else: return True
+
+		return False
 	#enddef
 
 	def registerCommand(self, name, function):
-		''' Register a command for use by a hotkey '''
+		''' Register a command that the user can bind with a hotkey '''
 		if name in self.commands:
 			log.warning('Registering over existing command name %s', name)
 		#endif
 
 		self.commands[name] = function
+
+		log.debug('Registered command {} for {}'.format(name, function))
+	#enddef
+
+	# TODO: These should directly use QKeySequence in the config
```

Comment:
You can do that with the logging function too. Using {} doesn't make it order-agnostic. log.debug('... %(name)s ...', name=string)